### PR TITLE
fix issue 18580 - std.conv.to!(const ubyte[])(void[]) should work

### PR DESCRIPTION
**Pull request by BBasile**
This PR allows the conversion from `void[]` to `immutable` and `const` `ubyte[]`. This will be useful for example to transform the result of `std.file.read` in `@safe` functions.

### CODE REVIEW

File: std/conv.d [hunk 1]
```diff
@@ -1555,6 +1556,17 @@ if (!isImplicitlyConvertible!(S, T) &&
     }
 }
 
+/**
+Void array to const array of ubyte conversion. Converting to a mutable array
+is not possible because source may contain pointers.
+ */
+private T toImpl(T, S)(S value) @trusted
+if (isArray!S && is(Unqual!(ElementType!S) == void) &&
+    isArray!T && is(ElementType!T == const ubyte) || is(ElementType!T == immutable ubyte))
```
Comment (BBasile) [inline std/conv.d]: Maybe that all basic types should work

File: std/conv.d [hunk 2]
```diff
@@ -1555,6 +1556,17 @@ if (!isImplicitlyConvertible!(S, T) &&
...
+{
+   return cast(const(ubyte)[]) value.dup;
```
Comment (ZombineDev) [inline std/conv.d]: `.to!T` should not allocate behind the user's back. I suggest you drop `.dup` and allow only conversion to `const ubyte[]`, which is safe if the input is not `shared`.

Comment (BBasile) [reply std/conv.d]: There was not real valid to dup anyway since the result is not modifiable.

**Comment by BBasile** [comment]
I'm gonna close this. It was only motivated for usage in D:YAML actually, which has to trust std.file.read.
Maybe it's better to make the file function safe.

**Pull request closed without merging**